primitives/core/src/traits.rs

```diff
@@ -16,16 +16,35 @@
 
 //! Shareable Substrate traits.
 
-use crate::{crypto::KeyTypeId, ed25519, sr25519};
+use crate::{
+	crypto::{KeyTypeId, CryptoTypePublicPair},
+	ed25519, sr25519,
+};
 
 use std::{
 	fmt::{Debug, Display},
+	collections::HashSet,
 	panic::UnwindSafe,
 	sync::Arc,
 };
 
 pub use sp_externalities::{Externalities, ExternalitiesExt};
 
+/// BareCryptoStore error
+#[derive(Debug)]
+pub enum Error {
```

Comment:
This `Error` is inside `traits` module, so it should have a more meaningful name, os that we don't have to rename it on import.
```suggestion
pub enum BareCryptoStoreError {
```